projects/week_2/Instructions.md

```diff
@@ -0,0 +1,275 @@
+
+Go to the ["src" folder](../../obstacle_detector/src), in the obstacle_detector package. Add two new files: "student_functions.cpp" and "student_functions.hpp".
+
+In the implementation file (student_functions.cpp), we need to include our new header file. Add an include statement at the top of the file to include the header file we just created.
+
+<details>
+<summary><b>Hint:</b> How to include a header</summary>
+<p>To include a header file named "my_header.hpp", we'd write the code below. Note that the use of quotes ("") instead of angle brackets (<>) indicates that this header is part of our current package.</p>
+<code>#include "my_header.hpp"</code>
+</details>
+
+In our new header file (student_functions.hpp), we need to add header guards. Add the three lines that define the header guards for this file. Remember that header guards look like this:
+
+```c++
+#ifndef STUDENT_FUNCTIONS_HPP  // Check that the unique macro has not already been defined
+#define STUDENT_FUNCTIONS_HPP  // Define the unique macro
+
+#endif  // Close the pre-processor if block
+```
+
+Finally, to include our new files into our package's build rules, we need to edit obstacle_detector's [CMakeLists.txt](../../obstacle_detector/CMakeLists.txt).
+
+Find the student code block in the call to `add_library`. Add our new implementation file, `src/student_code.cpp` to this function call.
+
+**Note:** We don't need to add our header file, because it gets compiled as part of the files that include it. Most C++ projects omit header files from the build rules unless there are special technical reasons that require them to be there.
+
+### 3.2 Declare functions
+
+Let's now declare the two functions we'll be creating for this project. First, because our functions will be using objects from the [OpenCV library](https://opencv.org/) for computer vision, we need to include a header from that library.
+
+Add an include directive for "opencv2/opencv.hpp". Remember that all code in our header file should go inside of the header guard if block (between `#define` and `#endif`).
+
+**Tip:** Because the OpenCV library is not part of our package, this include directive should use the angle bracket style.
+
+Next, add the declaration for our first function. This function should be named `FindColors`. Its return type should be `cv::Mat`, and it should take the following parameters:
+
+1. A constant `cv::Mat` named `input`
+1. A constant `cv::Scalar` named `range_min`
+1. A constant `cv::Scalar` named `range_max`
+
+Finally, add the declaration for our second function. This function should be named `ReprojectToGroundPlane`. Its return type should be `cv::Mat`, and it should take these parameters:
+
+1. A constant `cv::Mat` named `input`
+1. A constant `cv::Mat` named `homography`
+1. A constant `cv::Size` named `map_size`
+
+### 3.3 Implement `FindColors`
```

Comment:
If there are some bolded headings here that would be nice. This section is pretty long and hard to parse.

Reply:
I'll break it into two sections. One will cover the color space background and conversion and the other will do the range checking.

Reply:
I ended up also moving the couple of paragraphs about color spaces up into the background section.